application_store/db/queries/application/queries.py

```diff
@@ -240,14 +242,37 @@ def search_applications(**params):
     return found_apps
 
 
-def submit_application(application_id) -> Applications:
+def update_application_fields(existing_json_blob, new_json_blob) -> set:
+    field_map = {}
+    changed_fields = set()
+    for form in existing_json_blob["forms"]:
+        for section in form["questions"]:
+            for field in section["fields"]:
+                field_map[field["key"]] = field["answer"]
+
+    for form in new_json_blob["forms"]:
+        for section in form["questions"]:
+            for field in section["fields"]:
+                if field["answer"] != field_map.get(field["key"]):
+                    changed_fields.add(field["key"])
+                    if "history_log" in field:
+                        field["history_log"].append(
+                            {datetime.now(tz=timezone.utc).isoformat(): field_map[field["key"]]}
+                        )
+                    else:
+                        field["history_log"] = [{datetime.now(tz=timezone.utc).isoformat(): field_map[field["key"]]}]
+
+    return changed_fields
+
+
+def submit_application(application_id) -> Applications:  # noqa: C901
```

Comment:
Not the end of the world. But with the recent `update_application_fields` extraction, do we still break the C901  rule?